scripts/build_iso.sh

```
@@ -53,20 +74,85 @@ if [ -d "$WORK_DIR" ]; then
+    exit 1
+fi
+
+# Verify permissions
+if [ ! -x "$CONFIG_DIR/airootfs/usr/local/bin/pit" ]; then
+    echo -e "${RED}Error: pit wrapper is not executable${NC}"
+    exit 1
+fi
+
+THREADS=$(nproc)
+MEM_GB=$(free -g | awk '/^Mem:/{print $2}')
+WORK_DIR_ORIG="$WORK_DIR"
```

Shellcheck (0.10.0)

[warning] WORK_DIR_ORIG appears unused. Verify use (or export if used externally).

(SC2034)